control/tap-ctl-spawn.c

```diff
@@ -172,8 +174,42 @@ tap_ctl_get_child_id(int readfd)
 	return id;
 }
 
+/* Move the process to the nominated cgroup slice */
+static int
+tap_ctl_move_to_cgroup(int pid, const char *slice)
+{
+	FILE *f = NULL;
+	char *path;
+
+	if (!slice)
+		return -EINVAL;
+
+	path = malloc(FILENAME_MAX);
+	if (!path) {
+		EPRINTF("malloc failed: %d\n", errno);
+		return -errno;
+	}
+
+	snprintf(path, FILENAME_MAX, "/sys/fs/cgroup/cpu/%s/tasks", slice);
+	path[FILENAME_MAX-1] = '\0';
```

Comment:
@jjd27 I am not aware of any snprintf implementation that does not add a null char at the end, apart from Window's _snprintf that is spelled differently, anyway.
I just don't want to add coverity noise to our already noisy report. If you are aware of other
snprintf implementations that don't do that, please let me know and I will be more than heppy to reconsider.

@MarkSymsCtx FILENAME_MAX is supposed to be large enough to contain the string even though the implementation does not say if it is including null or not.
However, given any other macro like that (e.g. PATH_MAX) states explicitly that it must include
null termination, I would say that it is the same for FILENAME_MAX.
This means that if we do truncate, yes, fopen will fail because the file is not what we meant but
it would have failed anyway, given the file we passed cannot exist on the system because it would violate the max lenght, anyway.